src/NzbDrone.Core/Indexers/Definitions/Xthor/Xthor.cs

```diff
@@ -0,0 +1,470 @@
+                                UploadVolumeFactor = 1,
+                                DownloadVolumeFactor = torrent.Freeleech == 1 ? 0 : 1,
+                                Guid = guid,
+                                InfoUrl = details,
+                                DownloadUrl = link,
+                                TmdbId = torrent.Tmdb_id
+                            };
+
+                            return release;
+                        }));
+                }
+            }
+
+            return torrentInfos.ToArray();
+        }
+
+        private void CheckApiState(XthorError state)
+        {
+            // Switch on state
+            switch (state.Code)
+            {
+                case 0:
+                    // Everything OK
+                    break;
+                case 1:
+                    // Passkey not found
+                    throw new Exception("Passkey not found in tracker's database");
+                case 2:
+                    // No results
+                    break;
+                case 3:
+                    // Power Saver
+                    break;
+                case 4:
+                    // DDOS Attack, API disabled
+                    throw new Exception("Tracker is under DDOS attack, API disabled");
+                case 8:
+                    // AntiSpam Protection
+                    throw new Exception("Triggered AntiSpam Protection, please delay your requests !");
+                default:
+                    // Unknown state
+                    throw new Exception("Unknown state, aborting querying");
+            }
+        }
+
+        public Action<IDictionary<string, string>, DateTime?> CookiesUpdater { get; set; }
+    }
+
+    public class XthorSettingsValidator : AbstractValidator<XthorSettings>
+    {
+    }
+
+    public class XthorSettings : IIndexerSettings
+    {
+        private static readonly XthorSettingsValidator Validator = new XthorSettingsValidator();
+
+        public XthorSettings()
+        {
+            BaseUrl = "https://api.xthor.tk/";
+            Passkey = "";
+            FreeleechOnly = false;
+            Accent = 0;
+            NeedMultiReplacement = false;
+            MultiReplacement = "";
+            SubReplacement = "";
+            EnhancedAnime = true;
+            ByPassPageForTmDbid = true;
+            MaxPages = 1;
+        }
+
+        [FieldDefinition(1,
+            Label = "Url",
+            Type = FieldType.Select,
+            SelectOptionsProviderAction = "getUrls",
+            HelpText = "Site links")]
+        public string BaseUrl { get; set; }
+
+        [FieldDefinition(2,
+            Label = "Passkey",
+            Privacy = PrivacyLevel.Password,
+            Type = FieldType.Password,
+            HelpText = "Passkey to use for requests. (see https://xthor.tk/api_doc.php)")]
+        public string Passkey { get; set; }
+
+        [FieldDefinition(2,
+            Label = "Freeleech only",
```

Comment:
single line as well and 2 is being used twice here

Reply:
Fixed